### DISCUSSION TITLE
Re-get received messages

### DISCUSSION THREAD

**Discussion by adi-gor** [AsamK/signal-cli#537] [Q&A]
How to get messages which are already received to the linked device. I want to get all messages which I have received  till now from signal.

**Comment by AsamK** [comment]
You can't.
signal-cli doesn't store received messages and Signal doesn't provide a way to get messages from linked devices

**Reply by Devarsh-leo** [reply]
How does signal server decides what was the last message it sent? There might be some key/flag/counter stored in config file. Is there anything like that?

**Reply by AsamK** [reply]
No, once a message is received by a client, it is deleted from the server.
